src/error.rs

```diff
@@ -19,6 +44,6 @@ pub enum AsSliceError {
     Null,
 
     /// The output dtype does not match the data type of the array.
-    #[error("Desired output dtype does not match the data type of the array.")]
-    DtypeMismatch,
+    #[error("dtype mismatch: {0}")]
+    DtypeMismatch(String),
```

Comment:
I generally try to avoid carrying a runtime string as much as possible. The same error message can be obtained by just carrying the desired `Dtype` and actual `Dtype`

Reply:
I think some of the other errors types don't really need to carry a `String` either

Reply:
You made this one :D I just updated the message to be similar to the others

Reply:
I think its useful for actually giving context on what params may have been wrong and what we received. It should help a lot with debugging. Check out how the errors have been used so far.

Reply:
A potential alternative is 

```rust
pub enum AsSliceError {
    Null,
    DtypeMismatch {
        expecting: Dtype,
        found: Dtype,
    }
}
```

But let's leave it as it is for now and come back later

Reply:
I went ahead and changed it!